File: src/mysqlcookbook/recipes/metadata/table_exists.pl
#!/usr/bin/perl
# table_exists.pl

# Test for table existence.  Name a database and one or more
# tables on the command line.

use strict;
use warnings;
use Cookbook;

# Test whether a table exists.

# Caveat: table_exists() returns false if the table exists but you
# don't have privileges for accessing it.

#@ _TABLE_EXISTS_
sub table_exists
{
my ($dbh, $db_name, $tbl_name) = @_;

  return $dbh->selectrow_array (
                "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES
                WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?",
                undef, $db_name, $tbl_name);
}
#@ _TABLE_EXISTS_

@ARGV < 2 and die "Usage: $0 db_name tbl_name ...\n";
my $db_name = shift @ARGV;

my $dbh = Cookbook::connect ();

foreach my $tbl_name (@ARGV)
{
  printf "$tbl_name: %s\n",
         table_exists ($dbh, $db_name, $tbl_name)
           ? "exists"
           : "does not exist";
}

$dbh->disconnect ();

File: src/mysqlcookbook/recipes/metadata/table_exists.rb
#!/usr/bin/ruby
# table_exists.rb

# Test for table existence.  Name a database and one or more
# tables on the command line.

require "Cookbook"

# Test whether a table exists.

# Caveat: table_exists() returns false if the table exists but you
# don't have privileges for accessing it.

#@ _TABLE_EXISTS_
def table_exists(client, db_name, tbl_name)
  sth = client.prepare("SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES
                        WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?")
  return sth.execute(db_name, tbl_name).count > 0
end
#@ _TABLE_EXISTS_

begin
  client = Cookbook.connect
rescue Mysql2::Error => e
  puts "Could not connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

if ARGV.length < 2
  puts "Usage: table_exists.rb db_name tbl_name ..."
  exit(1)
end

db_name = ARGV.shift

ARGV.each do |tbl_name|
  puts "#{tbl_name}: " +
    (table_exists(client, db_name, tbl_name) ? "exists" : "does not exist")
end

client.close
